waspc/src/Wasp/ExternalConfig/PackageJson.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Wasp.ExternalConfig.PackageJson
+  ( PackageJson (..),
+    getDependencies,
+    getDevDependencies,
+    analyzePackageJsonContent,
+    findPackageJsonFile,
+  )
+where
+
+import Control.Monad.Except (ExceptT (ExceptT), runExceptT)
+import Data.Aeson (FromJSON)
+import qualified Data.Aeson as Aeson
+import Data.Map (Map)
+import qualified Data.Map as M
+import GHC.Generics (Generic)
+import StrongPath (Abs, Dir, File', Path', toFilePath)
+import Wasp.AppSpec.App.Dependency (Dependency)
+import qualified Wasp.AppSpec.App.Dependency as D
+import Wasp.Generator.Common
+  ( prismaVersion,
+    reactRouterVersion,
+  )
+import Wasp.Project.Common
+  ( CompileError,
+    WaspProjectDir,
+    findFileInWaspProjectDir,
+    packageJsonInWaspProjectDir,
+  )
+import Wasp.Util (maybeToEither)
+import qualified Wasp.Util.IO as IOUtil
+
+data PackageJson = PackageJson
+  { name :: !String,
+    dependencies :: !DependenciesMap,
+    devDependencies :: !DependenciesMap
+  }
+  deriving (Show, Generic)
+
+getDependencies :: PackageJson -> [Dependency]
+getDependencies packageJson = D.fromList $ M.toList $ dependencies packageJson
+
+getDevDependencies :: PackageJson -> [Dependency]
+getDevDependencies packageJson = D.fromList $ M.toList $ devDependencies packageJson
+
+type DependenciesMap = Map PackageName PackageVersion
+
+type PackageName = String
+
+type PackageVersion = String
+
+instance FromJSON PackageJson
+
+analyzePackageJsonContent :: Path' Abs (Dir WaspProjectDir) -> IO (Either [CompileError] PackageJson)
+analyzePackageJsonContent waspProjectDir = runExceptT $ do
```

Comment:
I've used a GPT suggestion to simplify the existing code and use the [`Except` monad transformer](https://hackage.haskell.org/package/mtl-2.3.1/docs/Control-Monad-Except.html#g:2) to make the code cleaner. Let me know if this looks okay.